Remotion/Security/Core/Metadata/EnumerationReflector.cs

```diff
@@ -70,8 +70,8 @@ public EnumValueInfo GetValue (Enum value, MetadataCache cache)
       {
         string name = value.ToString ();
         info = new EnumValueInfo (TypeUtility.GetPartialAssemblyQualifiedName (value.GetType ()), name, Convert.ToInt32 (value));
-        // TODO RM-7868: Add debug notnull assertion
-        FieldInfo fieldInfo = value.GetType ().GetField (name, BindingFlags.Static | BindingFlags.Public)!;
+        FieldInfo? fieldInfo = value.GetType ().GetField (name, BindingFlags.Static | BindingFlags.Public);
+        Assertion.DebugIsNotNull (fieldInfo, "fieldInfo must not be null");
```

Comment:
```suggestion
        Assertion.DebugIsNotNull (fieldInfo, "Field '{0}' was not found on type '{0}'.", name, value.GetType());
```

Reply:
@MichaelKetting `Assertion.DebugIsNotNull` does not have an overload taking a format string with arguments.

In 958ed4fb2f5c1c9de577f64b18a338e7b29d5520, I used `Assertion.DebugIsNotNull (fieldInfo, string.Format ("Field '{0}' was not found on type '{1}'.", name, value.GetType()))`, because string interpolation does not seem to be used in this project. Also ok like this?

Reply:
okay, so we have to add an overload, please. it's not a good thing to do either type of string-manipulation here because this could would be executed for relase-builds, too (I think) and thus result in cpu overhead.

Reply:
Makes sense. Separate ticket or within this task?

Reply:
separate ticket / PR please

Reply:
-> [RM-7883](https://re-motion.atlassian.net/browse/RM-7883])

Reply:
This one waits for the rebase on the other task so we can drop the string.Format.

Reply:
Having merged RM-7883 and rebased, I have implemented your suggestion from above.